language-support/ts/daml-ledger/index.test.ts

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Template, ContractId, Choice } from "@daml/types";
+import WS from "jest-websocket-mock";
+import Ledger from "./index";
+import { Event, CreateEvent } from "./index";
+import * as jtv from "@mojotech/json-type-validation";
+
+type Foo = {};
+
+const Foo: Template<Foo, undefined, "foo-id"> = {
+  templateId: "foo-id",
+  keyDecoder: () => jtv.constant(undefined),
+  decoder: () => jtv.object({}),
+  Archive: ({} as unknown) as Choice<Foo, {}, {}, undefined>,
+};
+
+const fooEvent = (coid: number): Event<Foo, undefined, "foo-id"> => {
+  return { created: fooCreateEvent(coid) };
+};
+
+const fooCreateEvent = (
+  coid: number
+): CreateEvent<Foo, undefined, "foo-id"> => {
+  return {
+    templateId: "foo-id",
+    contractId: coid.toString() as ContractId<object>,
+    signatories: [],
+    observers: [],
+    agreementText: "fooAgreement",
+    key: undefined,
+    payload: {},
+  };
+};
+
+const fooArchiveEvent = (coid: number): Event<Foo, undefined, "foo-id"> => {
+  return {
+    archived: {
+      templateId: "foo-id",
+      contractId: coid.toString() as ContractId<object>,
+    },
+  };
+};
+
+const mockOptions = {
+  token: "dummyToken",
+  httpBaseUrl: "http://localhost:5000/",
+  wsBaseUrl: "ws://localhost:4000/",
+};
+
+afterEach(() => {
+  WS.clean();
+});
+
+describe("streamQuery", () => {
+  // we only check that nothing crashes.
+  test("receive unknown message", async () => {
+    const server = new WS("ws://localhost:4000/v1/stream/query", {
+      jsonProtocol: true,
+    });
+    const ledger = new Ledger(mockOptions);
+    const stream = ledger.streamQuery(Foo);
+    await server.connected;
+    console.log("connected");
+    stream.on("change", (evs) => console.log(evs));
+    stream.on("close", (ev) => console.log(ev));
+    stream.off("change", (evs) => console.log(evs));
+    stream.off("close", (ev) => console.log(ev));
```

Comment:
Why would we remove the handlers before anything has happened?
```suggestion
```
These two lines can go everywhere else too.